src/library-authoring/common/context.tsx

```diff
@@ -49,7 +80,7 @@ export interface LibraryContextData {
   componentBeingEdited: string | undefined;
   openComponentEditor: (usageKey: string) => void;
   closeComponentEditor: () => void;
-}
+} & ComponentPickerType;
```

Comment:
The `type` change is not very pretty on this side, but I think it is beneficial, as we add many related states in the context. It makes things safer using the `useLibraryContext` hook on the components side.

i.e
```typescript
  const {
    componentPickerMode,
    onComponentSelected,
    addComponentToSelectedComponents,
    removeComponentFromSelectedComponents,
  } = useLibraryContext();

  if (!componentPickerMode) {
    // ❌ This throws a type error
    addComponentToSelectedComponents({ usageKey: 'blockId', blockType: 'blockType' });
  };

  if (componentPickerMode === 'single') {
    // ❌ This throws a type error
    addComponentToSelectedComponents({ usageKey: 'blockId', blockType: 'blockType' });
  };

  if (componentPickerMode === 'multiple') {
    // ✅ This works
    addComponentToSelectedComponents({ usageKey: 'blockId', blockType: 'blockType' });
  };
```

Reply:
I think that's nice. But I think we will soon need to split up `LibraryContext` because it's getting too big and messy. We can instead have `LibraryContext`, `SidebarContext`, and `ComponentPickerContext`, and it will clean up our tests a lot too. The `SidebarContext`, and `ComponentPickerContext` can both have defaults so they don't always have to be present in order to read their values. (e.g. `isSidebarOpen` can just be false in the default context, even if there's no `SidebarContextProvider` available)